File: 2442-number-of-arithmetic-triplets/number-of-arithmetic-triplets.cpp
class Solution {
public:
    int arithmeticTriplets(vector<int>& nums, int diff) {
        map<int, int> hashMap;

        for(auto it: nums) {
            if( hashMap.find(it) != hashMap.end() )
                continue;
            else {
                if( hashMap.find(it-diff) == hashMap.end() )
                    hashMap[it] = 1;
                else 
                    hashMap[it] = hashMap[it-diff]+1;
            }
        }

        int ans = 0;
        for(const auto& [key, val] : hashMap) {
            // cout<<key<<"  "<<val<<endl;
            if(val >= 3)
                ans++;
        }
        return ans;
    }
};

File: 2442-number-of-arithmetic-triplets/number-of-arithmetic-triplets.java
class Solution {
    public int arithmeticTriplets(int[] nums, int diff) {
        
        boolean[] hashMap = new boolean[201];
        int n = nums.length;
        int ans = 0;
        int prev1,prev2;

        for(int e : nums)
            hashMap[e] = true;

        for(int i=0+2; i<n; i++) {
            prev1 = nums[i] - diff;
            prev2 = nums[i] - diff - diff;
            if(prev1>=0 && hashMap[prev1] && prev2>=0 && hashMap[nums[i] - diff - diff])
                ans++;
        }

        return ans;
    }
}